core/src/main/java/org/jenkins/ui/icon/IconSet.java

```diff
@@ -92,7 +106,7 @@ public static String getSymbol(String name, String title, String tooltip, String
         }
 
         // Load symbol if it exists
-        InputStream inputStream = IconSet.class.getResourceAsStream("/images/symbols/" + translatedName + ".svg");
+        InputStream inputStream = classToLoadFrom.getResourceAsStream("/images/symbols/" + translatedName + ".svg");
```

Comment:
I'm wondering if we should add another arg to the spec which is plugin-$pluginname and then use uberclassloader to load it if pluginname is set?

Reply:
Would it also make sense to provide an API plugin for ionic icons in the same way I created an API plugin for Font Awesome icons? Then plugin authors would get automatically a simple way to use new, modern and consistent icons. Otherwise we might get a new mix of different icon styles again. Additionally, searching for consistent icons that use the correct SVG attributes might be cumbersome as well for plugin authors.

Reply:
I think that would need to core as they are used here?

not a separate API plugin?

I've got something mostly working for the above, will prob finish it up tomorrow morning and push

Reply:
> I think that would need to core as they are used here?
> not a separate API plugin?
 
Yes, that would be even better. But I would expect that such a library is too big for core so I did not even ask for it 😁 

> I've got something mostly working for the above, will prob finish it up tomorrow morning and push

😍

Reply:
by above I meant namespacing plugins properly not the api plugin but will see if I have time for htat

Reply:
I thought I had a nice clean plan, but the classloading isn't working 😢 

https://github.com/jenkinsci/jenkins/pull/6395/files#r835055749

Reply:
resolved now